hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/BlockItemWriter.java

```diff
@@ -18,11 +19,13 @@ public interface BlockItemWriter {
     void openBlock(long blockNumber);
 
     /**
-     * Writes a serialized item to the destination stream.
+     * Writes an item and/or its serialized bytes to the destination stream.
      *
+     * @param item the item to write
      * @param bytes the serialized item to write
      */
-    default void writePbjItem(@NonNull final Bytes bytes) {
+    default void writePbjItemAndBytes(@NonNull final BlockItem item, @NonNull final Bytes bytes) {
```

Comment:
This requires _both_ the block item _and_ the bytes, how will that improve performance?

Reply:
We pass the `BlockItem` so that in `createPublishStreamRequests()` method we work directly with the needed type in order to create the `PublishStreamRequests`.

The `Bytes` are used for the writer that writes the items to file on disk and since we already have the item serialized in a `ParallelTask` in `BlockStreamManagerImpl` I didn't want to duplicate it for the file writer (otherwise we can pass `BlockItem` to the file writer too).